DuckDuckGo/Feedback/VPNFeedbackSender.swift

```diff
@@ -0,0 +1,47 @@
+//
+//  VPNFeedbackSender.swift
+//  DuckDuckGo
+//
+//  Copyright © 2023 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Foundation
+import Core
+
+protocol VPNFeedbackSender {
+    func send(metadata: VPNMetadata, category: VPNFeedbackCategory, userText: String) async throws
+}
+
+struct DefaultVPNFeedbackSender: VPNFeedbackSender {
+
+    func send(metadata: VPNMetadata, category: VPNFeedbackCategory, userText: String) async throws {
+        let encodedUserText = userText.addingPercentEncoding(withAllowedCharacters: .alphanumerics) ?? userText
```

Comment:
I consider this one a bug in the macOS implementation - I recently fixed it. Instead, we should use the following character set, which avoids overly encoding the feedback:

```suggestion
        let encodedUserText = userText.addingPercentEncoding(withAllowedCharacters: .alphanumerics.union(.init(charactersIn: "-._~"))) ?? userText
```